### ISSUE TITLE
Tasks deduplication

### ISSUE THREAD

**Issue by klesh** [merico-dev/lake#827]
## Description
When user trigger pipeline with identical tasks list multiple times, devlake would run those duplicated tasks in parallel, which is a big waste and took longer time for each pipeline to complete

## Describe the solution you'd like
When creating pipeline, run dedupe check on existing tasks, and reuse those tasks.

## Has the Feature been Requested Before?
No

## Additional context
EE need this to prevent situation like:
1. User click on "sync jira" and EE send out "collect jira -> compute jira/gitlab metric"
2. User then click on "sync gitlab" and EE send out "collect gitlab -> compute jira/gitlab metric" right away
3. Causing jira/gitlab got computed twice

**Comment by joncodo** [comment]
This is handled in this taks: https://github.com/merico-dev/lake/issues/775

If a new task is triggered while another one of the same type is running, cancel all running tasks of the same type and only execute the new one.

**Comment by klesh** [comment]
@joncodo I suggest we reopen this one, and close #775 instead.
for automatical cancelling, we need to consider more factors.

**Comment by joncodo** [comment]
@klesh feel free to change anything you need. I trust your ideas :)

**Comment by klesh** [comment]
Now we had 2d array pipeline, this is no longer an issue. close for now.
